frontend/filtering.cpp

```diff
@@ -54,11 +54,15 @@ void MainWindow::on_groupButton_clicked(bool checked)
         model_->unGroupData();
         for (auto i = 0; i < ui_->runDataTable->horizontalHeader()->count(); ++i)
         {
-            if (!desiredHeader_.contains(
-                    ui_->runDataTable->horizontalHeader()->model()->headerData(i, Qt::Horizontal).toString()))
-            {
+            auto found = false;
+            foreach (const auto &header, desiredHeader_)
+                if (header.first == model_->headerData(i, Qt::Horizontal, Qt::UserRole).toString())
+                {
+                    found = true;
+                    break;
+                }
+            if (!found)
```

Comment:
Another potential `std::find_if` (but check my logic):
```suggestion
            auto it = std::find_if(desiredHeader_.begin(), desiredHeader_.end(),
                                   [&](const auto &data) { return data.first == model_->headerData(i, Qt::Horizontal, Qt::UserRole).toString(); });
            if (it == desiredHeader_.end())
```